krm-functions/lib/condkptsdk/sdk_populate_inventory.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package condkptsdk
+
+import (
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	kptfilelibv1 "github.com/nephio-project/nephio/krm-functions/lib/kptfile/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// populateInventory populates the inventory with the conditions and resources
+// related to the config
+func (r *sdk) populateInventory() error {
+	// To make filtering easier the inventory distinguishes global resources
+	// versus specific resources associated to a forInstance (specified through the SDK Config).
+	// To perform this filtering we use the concept of the forOwnerRef, which is
+	// an ownerReference associated to the forGVK
+	// A watchedResource matching the forOwnerRef is assocatiated to the specific
+	// forInventory context. If no match was found to the forOwnerRef the watchedResource is associated
+	// to the global context
+	var forOwnerRef *corev1.ObjectReference
+	// we assume the kpt file is always resource idx 0 in the resourcelist, the object is used
+	// as a reference to errors when we encounter issues with the condition processing
+	// since conditions are stored in the kptFile
+	//o := r.rl.Items.GetRootKptfile()
+
+	// We first run through the conditions to check if an ownRef is associated
+	// to the for resource objects. We call this the forOwnerRef
+	// When a forOwnerRef exists it is used to associate a watch resource to the
+	// inventory specific to the for resource or globally.
+	for _, c := range r.kptf.GetConditions() {
```

Comment:
This is my understanding, please correct me if I'm wrong.
Populate inventory method loops over all the condition and resources. It knows "which is the ownerRef" based on if conditions are present. 
For example in NFDeployment scenario, will the condition be present in the Kptfile? If not, then that resource relationship will not be tracked in the inventory to prepare ready context to call GenerateResourceFn. 

If the condition will be present? Which upstream function creates it?